pkg/pool-weighted/contracts/smart/InvestmentPool.sol

```diff
@@ -0,0 +1,271 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/ReentrancyGuard.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/WordCodec.sol";
+
+import "../BaseWeightedPool.sol";
+import "./WeightCompression.sol";
+
+/**
+ * @dev Weighted Pool with mutable weights, designed to support V2 Liquidity Bootstrapping
+ */
+contract InvestmentPool is BaseWeightedPool, ReentrancyGuard {
+    // The Pause Window and Buffer Period are timestamp-based: they should not be relied upon for sub-minute accuracy.
```

Comment:
Why is this comment here? Seems like it was probably copied from another contract but it's not relevant to this one.

Reply:
Yes; stale cut/paste. Case of staring at it too long and not seeing things :)

Reply:
Seems like it wasn't removed?

Reply:
Well now I'm really confused, because it looks like this comment is from a later branch somehow? Panicked that I checked into the wrong branch somehow (there are three branched off each other), but it looks right otherwise. Well, the code will catch up to it soon.

Reply:
@EndymionJkb I don't understand what you mean; the code will never catch up to this comment because the Pause Window and Buffer Period are implemented in BaseWeightedPool, no? The comment should be removed.

Reply:
Ok - now I finally understand. You're talking about the Pause window comment, not the dev comment on top! The highlighted section included both, and I was looking at the top one :) Yes, will remove.